lib/utils/bit_list.ex

```diff
@@ -56,28 +73,41 @@ defmodule LambdaEthereumConsensus.Utils.BitList do
 
   def length_of_bitlist(bitlist) when is_binary(bitlist) do
     bit_size = bit_size(bitlist)
-    <<_::size(bit_size - 8), last_byte>> = bitlist
-    bit_size - leading_zeros(<<last_byte>>) - 1
+    <<_::size(bit_size - @bits_per_byte), last_byte>> = bitlist
+    bit_size - leading_zeros(<<last_byte>>) - @bits_in_sentinel_bit
   end
 
-  defp leading_zeros(<<1::1, _::7>>), do: 0
-  defp leading_zeros(<<0::1, 1::1, _::6>>), do: 1
-  defp leading_zeros(<<0::2, 1::1, _::5>>), do: 2
-  defp leading_zeros(<<0::3, 1::1, _::4>>), do: 3
-  defp leading_zeros(<<0::4, 1::1, _::3>>), do: 4
-  defp leading_zeros(<<0::5, 1::1, _::2>>), do: 5
-  defp leading_zeros(<<0::6, 1::1, _::1>>), do: 6
-  defp leading_zeros(<<0::7, 1::1>>), do: 7
+  defp leading_zeros(<<@sentinel_bit::@bits_in_sentinel_bit, _::7>>), do: 0
+  defp leading_zeros(<<0::1, @sentinel_bit::@bits_in_sentinel_bit, _::6>>), do: 1
+  defp leading_zeros(<<0::2, @sentinel_bit::@bits_in_sentinel_bit, _::5>>), do: 2
+  defp leading_zeros(<<0::3, @sentinel_bit::@bits_in_sentinel_bit, _::4>>), do: 3
+  defp leading_zeros(<<0::4, @sentinel_bit::@bits_in_sentinel_bit, _::3>>), do: 4
+  defp leading_zeros(<<0::5, @sentinel_bit::@bits_in_sentinel_bit, _::2>>), do: 5
+  defp leading_zeros(<<0::6, @sentinel_bit::@bits_in_sentinel_bit, _::1>>), do: 6
+  defp leading_zeros(<<0::7, @sentinel_bit::@bits_in_sentinel_bit>>), do: 7
   defp leading_zeros(<<0::8>>), do: 8
 
   @spec remove_trailing_bit(binary()) :: bitstring()
-  defp remove_trailing_bit(<<1::1, rest::7>>), do: <<rest::7>>
-  defp remove_trailing_bit(<<0::1, 1::1, rest::6>>), do: <<rest::6>>
-  defp remove_trailing_bit(<<0::2, 1::1, rest::5>>), do: <<rest::5>>
-  defp remove_trailing_bit(<<0::3, 1::1, rest::4>>), do: <<rest::4>>
-  defp remove_trailing_bit(<<0::4, 1::1, rest::3>>), do: <<rest::3>>
-  defp remove_trailing_bit(<<0::5, 1::1, rest::2>>), do: <<rest::2>>
-  defp remove_trailing_bit(<<0::6, 1::1, rest::1>>), do: <<rest::1>>
-  defp remove_trailing_bit(<<0::7, 1::1>>), do: <<0::0>>
+  defp remove_trailing_bit(<<@sentinel_bit::@bits_in_sentinel_bit, rest::7>>), do: <<rest::7>>
```

Comment:
I think this adds too much noise. We should remove the `@sentinel_bit` and `@bits_in_sentinel_bit` constants, and add this to the documentation either via comments or function/module docs (we can leave that for another PR)